packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -32,18 +34,42 @@ const FIRST_OBJECT_ID = 50n;
 const FIRST_PROMISE_ID = 60n;
 const FIRST_DEVICE_ID = 70n;
 
+// TODO: we export this from vatKeeper.js, and import it from
+// kernelKeeper.js, because both files need it, and we want to avoid
+// an import cycle (kernelKeeper imports other things from vatKeeper),
+// but it really wants to live in kernelKeeper not vatKeeper
+export const DEFAULT_REAP_DIRT_THRESHOLD_KEY =
+  'kernel.defaultReapDirtThreshold';
+
 /**
  * Establish a vat's state.
  *
  * @param {*} kvStore  The key-value store in which the persistent state will be kept
  * @param {*} transcriptStore  Accompanying transcript store
  * @param {string} vatID The vat ID string of the vat in question
  * TODO: consider making this part of makeVatKeeper
+ * @param {SourceOfBundle} source
+ * @param {RecordedVatOptions} options
  */
-export function initializeVatState(kvStore, transcriptStore, vatID) {
+export function initializeVatState(
+  kvStore,
+  transcriptStore,
+  vatID,
+  source,
+  options,
+) {
+  assert(source);
+  assert('bundle' in source || 'bundleName' in source || 'bundleID' in source);
+  assert(options);
+  assert.typeof(options, 'object');
+  assert(options.workerOptions, `vat ${vatID} options missing workerOptions`);
```

Comment:
Yeah.. that first option doesn't *look* like a simplification.. lotta lines there :). And that shape is already being expressed in typescript, I'm not sure I want to duplicate the form. The runtime check was mostly just to catch some API changes (adding a new argument) that tripped me up a  few years ago.

I'm happy to go with the second one, although with an assert instead of `goodthing || Fail`, for my idiosyncratic reasons listed above.

Reply:
> that first option doesn't _look_ like a simplification.. lotta lines there :)

Yes, but half of them belong in endo.

> that shape is already being expressed in typescript, I'm not sure I want to duplicate the form.

A very fair point... I think we're making progress on mechanical derivation, but AFAIK we aren't there yet.

> I'm happy to go with the second one, although with an assert instead of `goodthing || Fail`, for my idiosyncratic reasons listed above.

:+1: